drivers/adc/Kconfig.stm32

```diff
@@ -0,0 +1,155 @@
+# Kconfig - ADC configuration options
+
+#
+# Copyright (c) 2019 Intel Corporation
+# Copyright (c) 2019 Endre Karlson
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ADC_STM32
+	bool "STM32 ADC driver"
+	depends on SOC_FAMILY_STM32
+	help
+		Enable the driver implementation for the stm32xx ADC
+
+if ADC_STM32
+
+	menu ADC_0
```

Comment:
Rename to ADC1

Reply:
I really would like to do this while kconfig system of adc has a minor problem against this, which I described in #13501 . I will change them all to ADC_1 but then this should be merged after that PR.